vault/logical_system.go

```diff
@@ -2435,14 +2435,21 @@ func (b *SystemBackend) handlePoliciesSet(policyType PolicyType) framework.Opera
 			return nil, err
 		}
 
+		name := data.Get("name").(string)
 		policy := &Policy{
-			Name:      strings.ToLower(data.Get("name").(string)),
+			Name:      strings.ToLower(name),
 			Type:      policyType,
 			namespace: ns,
 		}
 		if policy.Name == "" {
 			return logical.ErrorResponse("policy name must be provided in the URL"), nil
 		}
+		if name != policy.Name {
+			if resp == nil {
+				resp = &logical.Response{}
+			}
```

Comment:
```suggestion
			resp = &logical.Response{}
```

I think `resp` is always `nil` at this point